crates/walrus-core/src/encoding/quilt_encoding.rs

```diff
@@ -503,11 +1052,68 @@ impl<'a> QuiltConfigApi<'a, QuiltVersionV1> for QuiltConfigV1 {
     }
 }
 
+struct MergeIterator<'a> {
```

Comment:
Question: Could you explain the purpose of this additional struct? (Why) do we need this somewhat complicated construction to just write some data to columns?